db/fixtures/ae_datastore/ManageIQ/Cloud/Orchestration/Provisioning/StateMachines/Methods.class/__methods__/check_provisioned.rb

```diff
@@ -1,70 +1,113 @@
 #
-# Description: This method checks to see if the stack has been provisioned
+# Description: This class checks to see if the stack has been provisioned
 #   and whether the refresh has completed
 #
+module ManageIQ
+  module Automate
+    module Cloud
+      module Orchestration
+        module Provisioning
+          module StateMachines
+            class CheckProvisioned
+              def initialize(handle = $evm)
+                @handle = handle
+              end
 
-def refresh_provider(service)
-  provider = service.orchestration_manager
+              def main
+                task = @handle.root["service_template_provision_task"]
+                service = task.try(:destination)
```

Comment:
We want to test whether service exists and is a type of ServiceOrchestration. Log and exit if not.